src/features/user/Profile/components/RedeemModal.tsx

```diff
@@ -266,17 +283,39 @@ export default function RedeemModal({
           <div className={styles.modal}>
             {codeValidated && validCodeData ? (
               <>
-                <div className={styles.codeTreeCount}>
+                <div
+                  className={styles.codeTreeCount}
+                  style={{ fontSize: '2rem' }}
+                >
                   {getFormattedNumber(
                     i18n.language,
-                    Number(validCodeData.treeCount)
+                    Number(validCodeData.units)
                   )}
                   <span>
                     {t('common:tree', {
-                      count: Number(validCodeData.treeCount),
+                      count: Number(validCodeData.units),
                     })}
                   </span>
                 </div>
+                <div style={{ position: 'relative' }}>
+                  <button
+                    className={styles.crossButton}
+                    style={{ top: '-57px', right: '-57px' }}
+                    onClick={closeRedeem}
+                  >
+                    <CancelIcon color={styles.primaryFontColor} />
+                  </button>
+                  <span className={styles.codeTreeCount}>
+                    {t('redeem:successfullyRedeemed')}
+                  </span>
+                </div>
+
+                <button
+                  className="primaryButton redeemAnotherCode"
+                  onClick={afterSuccessfulOfOneCode}
+                >
+                  {t('redeem:redeemAnotherCode')}
+                </button>
```

Comment:
If possible, I think we could have a common structure here and from lines 333-365